### ISSUE TITLE
fix: Autocomplete / Combobox hangs if many items are bound

### ISSUE THREAD

**Issue by martinbu** [microsoft/fluentui-blazor#1638]
If many items are bound to a FluentAutocomplete or FluentCombobox all items gets rendered. In the newest dev branch the items are even "rendered" before the combobox is opened.
This causes - depending on the items count - a lag of multiple seconds when opening the page or in a previous version when opening the combobox (list).

Is there a possibility to only render those items that are visible like the virtualization in the FluentDataGrid?

An example that reproduces this issue with the FluentCombobox can be found [here](https://github.com/martinbu/fluentui-blazor/commit/e2cd1ac0eca210d51e811fa3dc7dd4a91ea068c0).

**Comment by vnbaaij** [comment]
No, we can't do virtualuzation here as the (Fluent) options does not support that.

How many items are we talking about?

**Comment by martinbu** [comment]
At the moment I have a list with about 800 items. But because these controls have the possibilities to search/filter the bound items count could be even much much longer for other use cases...

**Comment by martinbu** [comment]
That is true, but if no search text is specified I want to "preserve" the possibility to scroll through the complete list. A user that does not know that he/she can search with the keyboard will then only see e.g. the first 10 items which can be confusing.

Additionally, the FluentAutocomplete has at the moment no possibility to "look like" a e.g. FluentCombobox with a simple one item selected view. On the other hand I cannot use a FluentCombobox because it has insufficient search capabilities like stated here #1639

**Comment by martinbu** [comment]
When I look at the two Components FluentAutoComplete and FluentCombobox I think it is much easier to add the functionalities
1. List Virtualization
2. Single Select option --> Autocomplete should look like a Combobox in single select mode

The virtualization part could be implemented with [ASP.NET Core Razor component virtualization](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/virtualization?view=aspnetcore-8.0) because the options are rendered with a foreach loop in the FluentAutocomplete.razor page.

For the "look like a combobox" part, an addtional option could be added which changes the appearance.

What do you think?

**Comment by dvoituron** [comment]
The FluentCombobox is a functionally similar representation of the same HTML component. The data to be searched must be present in the HTML page already loaded. This is the main reason why we've added the FluentAutocomplete, which allows you to launch a search on the "server" to obtain some results.
The constraint is that you have to start a search to display data.

But indeed, the FluentCombobox implementation uses a loop to fill in the options. I'd be curious to see if it's possible to place a List Virtualization there. In my opinion, there will be some difficulties in filtering the data.
Feel free to adapt the code and create a PR if the results are positive.

**Comment by vnbaaij** [comment]
> That is true, but if no search text is specified I want to "preserve" the possibility to scroll through the complete list. A user that does not know that he/she can search with the keyboard will then only see e.g. the first 10 items which can be confusing.
Agree that it can be confusing to see a limited set but can we also agree that presenting 800 items in a list is just not good user interface design.

> Additionally, the FluentAutocomplete has at the moment no possibility to "look like" a e.g. FluentCombobox with a simple one item selected view. On the other hand I cannot use a FluentCombobox because it has insufficient search capabilities like stated here #1639
They are differennt components so yes, there is indeed no option to make one like the other because they are not the same (on purpose). As the docs say, `<FluentCombobox>` wraps the `<fluent-combobox>` element, a web component implementation of a combobox. There is alink to the W3C specs of a combobox in there as well. As it wraps the web component, we are sort of limited in what we can change and adjust there.

**Comment by martinbu** [comment]
@vnbaaij I understand. Using FluentCombobox is NOT the way to go. But extending FluentAutocomplete may be the right choice.

Yes I a agree that presenting 800 items in a list may not be a good user interface design. But the point is that we have a scrollbar. So actually visible are only e.g. 10-20 items. Then the user can type to filter or if the user do not want to type the user can scroll.

For example, Windows is doing the same for language. And this is also my use case. I have a list with all languages including region specific languages.

Windows language selection:
![grafik](https://github.com/microsoft/fluentui-blazor/assets/1791936/47a3c35c-02db-4d67-b51a-bc7834146aad)

@dvoituron Ok, thank you for your feedback. I will try to add prototype implementation. Then we can discuss the result.

**Comment by vnbaaij** [comment]
Closing this one then for now. We can re-open once a PR is in place (or maybe better to create a new issue then)
